File: src/codes/Users_Manual/Tutorial/move_focus/soft.cpp
//~#include <chrono>
//~#include <thread>
//~#include "autd3.hpp"
//~#include "autd3/link/nop.hpp"
//~int main() {
//~auto autd = autd3::Controller::open({autd3::AUTD3{.pos =
//~autd3::Point3::origin(), .rot = autd3::Quaternion::Identity(),}},
//~autd3::link::Nop{});
autd.send(autd3::Sine(150 * autd3::Hz, autd3::SineOption{}));

const autd3::Point3 center = autd.center() + autd3::Vector3(0., 0., 150.);
while (true) {
  autd.send(
      autd3::Focus{center + autd3::Vector3(20., 0., 0.), autd3::FocusOption{}});

  std::this_thread::sleep_for(std::chrono::milliseconds(1000));

  autd.send(
      autd3::Focus{center - autd3::Vector3(20., 0., 0.), autd3::FocusOption{}});

  std::this_thread::sleep_for(std::chrono::milliseconds(1000));
}
//~return 0;
//~}

File: src/codes/Users_Manual/Tutorial/move_focus/soft.rs
# use autd3::prelude::*;
# fn main() -> Result<(), Box<dyn std::error::Error>> {
# let mut autd = Controller::open([AUTD3::default()], Nop::new())?;
autd.send(Sine {
    freq: 150 * Hz,
    option: SineOption::default(),
})?;

let center = autd.center() + Vector3::new(0.0, 0.0, 150. * mm);
loop {
    autd.send(Focus {
        pos: center + Vector3::new(20. * mm, 0.0, 0.0),
        option: FocusOption::default(),
    })?;

    std::thread::sleep(std::time::Duration::from_secs(1));

    autd.send(Focus {
        pos: center - Vector3::new(20. * mm, 0.0, 0.0),
        option: FocusOption::default(),
    })?;

    std::thread::sleep(std::time::Duration::from_secs(1));
}
# }